### PR TITLE
refactor(types): improve of type assertions

### PR DESCRIPTION
**Pull request by webfansplz**
improve of type assertions

### CODE REVIEW

**Comment by posva** [comment]
Thanks, FYI, it's normal for the e2e to fail until 3.2.0 is released. Let me know when you are done 🙂

**Comment by webfansplz** [comment]
> Thanks, FYI, it's normal for the e2e to fail until 3.2.0 is released. Let me know when you are done 🙂

I have finished～

File: src/matcher/index.ts [hunk 1]
```diff
@@ -370,13 +371,13 @@ function normalizeRecordProps(
 ): Record<string, _RouteRecordProps> {
   const propsObject = {} as Record<string, _RouteRecordProps>
   // props does not exist on redirect records but we can set false directly
-  const props = (record as any).props || false
+  const props = (record as RouteRecordSingleView).props || false
```
Comment (posva) [inline src/matcher/index.ts:374]:
```suggestion
  const props = (record as Exclude<RouteRecordRaw, RouteRecordRedirect>).props || false
```

File: src/router.ts [hunk 2]
```diff
@@ -464,11 +464,11 @@ export function createRouter(options: RouterOptions): Router {
         __DEV__ &&
         'params' in rawLocation &&
         !('name' in rawLocation) &&
-        Object.keys((rawLocation as any).params).length
+        Object.keys((rawLocation as RouteLocation).params).length
       ) {
         warn(
           `Path "${
-            (rawLocation as any).path
+            (rawLocation as RouteLocation).path
```
Comment (posva) [inline src/router.ts:471]: This one is not true, let's revert it to any, it doesn't matter

**Review by posva — CHANGES_REQUESTED** [review]
Thanks!

**Comment by webfansplz** [comment]
@posva thank you for your review 😊

File: src/router.ts [hunk 3]
```diff
@@ -464,7 +464,7 @@ export function createRouter(options: RouterOptions): Router {
         __DEV__ &&
         'params' in rawLocation &&
         !('name' in rawLocation) &&
-        Object.keys((rawLocation as any).params).length
+        Object.keys((rawLocation as RouteLocation).params).length
```
Comment (posva) [inline src/router.ts:467]:
```suggestion
        Object.keys((rawLocation as any).params).length
```

**Pull request merged**